### ISSUE TITLE
Svelte Scoped bug: Preflights overwrite classes

### ISSUE THREAD

**Issue by gottlike** [unocss/unocss#2944]
### UnoCSS version

v0.54.1

### Describe the bug

When using preflights that specify a rule for a tag (e.g. as part of a style reset CSS), those rules overwrite the styling added by classes. I am using the Svelte Scoped plugin to create Web Components and ran into this issue after injecting a style reset CSS.

I created a StackBlitz reproduction here: https://stackblitz.com/edit/unocss-unocss-d7aipc?file=uno.config.ts

### Reproduction

https://stackblitz.com/edit/unocss-unocss-d7aipc?file=uno.config.ts

### System Info

Irrelevant

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by jacob-8** [comment]
You've set up both @unocss/svelte-scoped/preprocess and @unocss/svelte-scoped/vite so that right there is bound to give some conflicts. Just choose one. In this case of building web components I'm assuming you'll want the former. It's only been battle tested for building Svelte libraries and not web components. It may still work just as well for web components but no experience there. Can you make a new repro using just the preprocessor and have an example of a web component in it?

Note that UnoCSS does have a regular Vite plugin with web component mode though I don't know it's usefulness level with Svelte.

**Comment by gottlike** [comment]
@jacob-8, I used the official UnoCSS svelte-scoped example as base for the repro. If it's not advisable to use both libraries together, the example code should probably be modified accordingly.

I edited the repro to be more in line with what I actually use right now and specifying the web component. I also removed `@unocss/svelte-scoped/vite` from `svelte.config.js`. The result is the same, however: https://stackblitz.com/edit/unocss-unocss-lyj1wn?file=svelte.config.js
